modules/apps/portal-language/portal-language-override-service/src/main/java/com/liferay/portal/language/override/internal/util/PLOLanguageOverrideProviderUtil.java

```diff
@@ -0,0 +1,120 @@
+
+import com.liferay.osgi.util.service.Snapshot;
+import com.liferay.petra.concurrent.DCLSingleton;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.language.override.model.PLOEntry;
+import com.liferay.portal.language.override.service.PLOEntryLocalServiceUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+/**
+ * @author Regisson Cesar
+ */
+public class PLOLanguageOverrideProviderUtil {
+
+	public static void add(PLOEntry ploEntry) {
+		_add(_ploEntriesMapDCLSingleton.getSingleton(_supplier), ploEntry);
+	}
+
+	public static String encodeKey(long companyId, String languageId) {
+		return StringBundler.concat(companyId, StringPool.POUND, languageId);
+	}
+
+	public static Supplier<Map<String, HashMap<String, String>>> getSupplier() {
+		return _supplier;
+	}
+
+	public static void remove(PLOEntry ploEntry) {
+		Map<String, HashMap<String, String>> ploEntriesMap =
+			_ploEntriesMapDCLSingleton.getSingleton(_supplier);
+
+		ploEntriesMap.computeIfPresent(
+			encodeKey(ploEntry.getCompanyId(), ploEntry.getLanguageId()),
+			(key, value) -> {
+				value.remove(ploEntry.getKey());
+
+				if (value.isEmpty()) {
+					return null;
+				}
+
+				return value;
+			});
+	}
+
+	public static void update(PLOEntry ploEntry) {
+		Map<String, HashMap<String, String>> ploEntriesMap =
+			_ploEntriesMapDCLSingleton.getSingleton(_supplier);
+
+		ploEntriesMap.computeIfPresent(
+			encodeKey(ploEntry.getCompanyId(), ploEntry.getLanguageId()),
+			(key, value) -> {
+				value.put(ploEntry.getKey(), ploEntry.getValue());
+
+				return value;
+			});
+	}
+
+	private static void _add(
+		Map<String, HashMap<String, String>> ploEntriesMap, PLOEntry ploEntry) {
+
+		ploEntriesMap.compute(
+			encodeKey(ploEntry.getCompanyId(), ploEntry.getLanguageId()),
+			(key, value) -> {
+				if (value == null) {
+					value = new HashMap<>();
+				}
+
+				value.put(ploEntry.getKey(), ploEntry.getValue());
+
+				return value;
+			});
+	}
+
+	private static Map<String, HashMap<String, String>> _createPLOEntriesMap() {
+		Map<String, HashMap<String, String>> ploEntriesMap =
+			new ConcurrentHashMap<>();
+
+		CompanyLocalService companyLocalService =
+			_companyLocalServiceSnapshot.get();
+
+		companyLocalService.forEachCompanyId(
+			companyId -> {
+				for (PLOEntry ploEntry :
+						PLOEntryLocalServiceUtil.getPLOEntries(companyId)) {
+
+					_add(ploEntriesMap, ploEntry);
+				}
+			});
+
+		return ploEntriesMap;
+	}
+
+	private static final Snapshot<CompanyLocalService>
+		_companyLocalServiceSnapshot = new Snapshot<>(
+			PLOLanguageOverrideProviderUtil.class, CompanyLocalService.class);
+	private static final DCLSingleton<Map<String, HashMap<String, String>>>
+		_ploEntriesMapDCLSingleton = new DCLSingleton<>();
```

Comment:
@regiCesar21 You have 2 `DCLSingleton`s, a `_ploEntriesMapDCLSingleton` here, and a `_ploEntriesMapDCLSingleton` in the original `PLOLanguageOverriderProvider`. The `_supplier` just blindly creates a new map, so I believe this `Util` and the `Provider` will have 2 different maps.

Please fix it.